### ISSUE TITLE
validate only

### ISSUE THREAD

**Issue by starbug42** [emreacar/Stripe-Payment-For-Opencart-3.x#19]
after hitting confirm, says validating, but ten nothing happens. no console log errors. oc 3.0.2

**Comment by jp-senthilkumar** [comment]
yes. I am also facing this same issue. If I click the "confirm order" button it is validating but nothing will happen next. I am struck up in this issue.  Kindly someone revert this issue.

**Comment by jp-senthilkumar** [comment]
Dear All,

I found the solution for this. The problem is the database table of " oc_stripe_order, oc_stripe_card and oc_stripe_customer " not created automatically in database. So I created these three tables in my Mysql database inside phpmyadmin then the issue has been resolved and now this extension is working fine. 

Check your database if the above three tables not created then Create the Tables manually in PhpmyAdmin as mentioned below : 

CREATE TABLE IF NOT EXISTS `oc_stripe_order` (
			`stripe_order_id` varchar(255) NOT NULL,
			`order_id` int(11) NOT NULL DEFAULT '0',
			`environment` varchar(5) NOT NULL DEFAULT 'test',
			PRIMARY KEY (`stripe_order_id`)
		) ENGINE=MyISAM DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci

------------------------------------------------------

CREATE TABLE IF NOT EXISTS `oc_stripe_card` (
			  `stripe_card_id` varchar(255) NOT NULL,
			  `customer_id` INT(11) NOT NULL,
			  `last_four` VARCHAR(4) NOT NULL,
			  `exp_year` VARCHAR(5) NOT NULL,
			  `exp_month` VARCHAR(5) NOT NULL,
			  `brand` VARCHAR(50) NOT NULL,
			  `environment` varchar(5) NOT NULL DEFAULT 'test',
			  PRIMARY KEY (`stripe_card_id`)
			) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci;

------------------------------------------------------

CREATE TABLE IF NOT EXISTS oc_stripe_customer (
customer_id INT(11) NOT NULL,
stripe_customer_id varchar(255) NOT NULL,
environment varchar(5) NOT NULL DEFAULT 'test',
PRIMARY KEY (stripe_customer_id)
) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci

------------------------------------------------------

I dont know why these tables are not created automatically. This extension developer should look into this issue and resolve it soon.

**Comment by emreacar** [comment]
Hi @jpsnifty , This database installation already exist in install() method. When plugin installed firstly method create theese tables automatically. 

But on some opencarts these install method didn't call from the plugin installations. I dont't know why..

https://github.com/emreacar/Stripe-Payment-For-Opencart-3.x/blob/master/upload/admin/model/extension/payment/stripe.php

**Comment by lamellama** [comment]
> Dear All,
> 
> I found the solution for this. The problem is the database table of " oc_stripe_order, oc_stripe_card and oc_stripe_customer " not created automatically in database. So I created these three tables in my Mysql database inside phpmyadmin then the issue has been resolved and now this extension is working fine.
> 
> Check your database if the above three tables not created then Create the Tables manually in PhpmyAdmin as mentioned below :
> 
> CREATE TABLE IF NOT EXISTS `oc_stripe_order` (
> `stripe_order_id` varchar(255) NOT NULL,
> `order_id` int(11) NOT NULL DEFAULT '0',
> `environment` varchar(5) NOT NULL DEFAULT 'test',
> PRIMARY KEY (`stripe_order_id`)
> ) ENGINE=MyISAM DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci
> 
> CREATE TABLE IF NOT EXISTS `oc_stripe_card` (
> `stripe_card_id` varchar(255) NOT NULL,
> `customer_id` INT(11) NOT NULL,
> `last_four` VARCHAR(4) NOT NULL,
> `exp_year` VARCHAR(5) NOT NULL,
> `exp_month` VARCHAR(5) NOT NULL,
> `brand` VARCHAR(50) NOT NULL,
> `environment` varchar(5) NOT NULL DEFAULT 'test',
> PRIMARY KEY (`stripe_card_id`)
> ) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci;
> 
> CREATE TABLE IF NOT EXISTS `" . DB_PREFIX . "stripe_customer` (
> `customer_id` INT(11) NOT NULL,
> `stripe_customer_id` varchar(255) NOT NULL,
> `environment` varchar(5) NOT NULL DEFAULT 'test',
> PRIMARY KEY (`stripe_customer_id`)
> ) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci;
> 
> I dont know why these tables are not created automatically. This extension developer should look into this issue and resolve it soon.

This worked great but just to save anyone else 10mins, the last query has an error it should be:
CREATE TABLE IF NOT EXISTS oc_stripe_customer (
customer_id INT(11) NOT NULL,
stripe_customer_id varchar(255) NOT NULL,
environment varchar(5) NOT NULL DEFAULT 'test',
PRIMARY KEY (stripe_customer_id)
) ENGINE=MyISAM DEFAULT COLLATE=utf8_general_ci;
